### ISSUE TITLE
App resets when switching windows on android.

### ISSUE THREAD

**Issue by TacoError** [fyne-io/fyne#2989]
### Describe the bug:
I run my app, which requires a task to run whether in or out of the app once started.

Once i leave the window to goto google on the same phone, and go back to that window, the app is reset.

### To Reproduce:
Steps to reproduce the behaviour:
1: do something that makes the app look like it didnt just start up
2: goto a diff window
3: go back to the app
4: app is reset back

### Device (please complete the following information):
 - **OS:** Android
 - **Version:** 11
 - **Go version:** 1.181
 - **Fyne version:** 2.4.1

**Comment by andydotxyz** [comment]
Can you please see if anything appears in the log (the usual 'android logcat' should work).
This is not normal behaviour and it may be hitting a fault or memory issue.
Also it might help to define "reset" - is it like the app has restarted completely or are only some elements impacted?

**Comment by andydotxyz** [comment]
I think we need to see the logs (`adb logcat`) I think that perhaps the OS is killing your app.
Mobile does not allow background goroutine to run indefinitely, we need to find a way to add an API that provides guaranteed background processes I think.

**Comment by habitualdev** [comment]
After doing some testing (context - Samsung Note10 Ultra) I was able to replicate both the issue and my current solution. 

I needed my app to run in the background for notifications. Upon initial install, if I switched the app into the background, it would suspend everything. My current fix is to change the app's battery policy to unrestricted. This makes the app run in the background as intended. 

I did this in the phone's UI, but I imagine there's a programmatic way to do this with fyne-cross.
